pmd-java/src/main/java/net/sourceforge/pmd/lang/java/symbols/internal/asm/GenericSigBase.java

```diff
@@ -81,7 +81,11 @@ protected final void ensureParsed() {
 
     protected abstract boolean postCondition();
 
-    protected abstract boolean isGeneric();
+    protected abstract int getTypeParameterCount();
```

Comment:
Each GenericSigBase impl must provide a implementation for getTypeParameterCount()